### PR TITLE
fix(select): adjust clear button

### PR DESCRIPTION
**Pull request by cooper-joe**
Implements [UX-154](https://dhis2.atlassian.net/browse/UX-154)

---

### Description

This PR changes the `SingleSelect` and `MultiSelect` clear button to use an icon button rather than a `Button` with text label.

`clearText` prop is used to populate the `Tooltip` that appears on hover.

---

### Checklist

-   [x] API docs are generated
-   [x] Tests were added
-   [x] Storybook demos were added

_All points above should be relevant for feature PRs. For bugfixes, some points might not be relevant. In that case, just check them anyway to signal the work is done._

---

### Screenshots

Before: 
image
image

After:
image
image

[UX-154]: https://dhis2.atlassian.net/browse/UX-154?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: components/select/src/single-select/input.js
Comment (Mohammer5) [inline components/select/src/single-select/input.js:27]: This doesn't appear to be correct, see here: https://github.com/dhis2/ui/blob/4643200c1adfdd40223e0b4d04b144b7d9a25a4a/components/button/src/button/button.js#L39

* Callback is passed to `InputClearButton`
* `InputClearButton` passes the callback to `Button`
* `Button` calls the onClick callback with the event as second argument

**Review by Mohammer5 — APPROVED** [review]
~~This is actually a change request, but I won't be working next week, so approving so I don't block this PR~~

EDIT: :+1:

File: components/select/src/single-select/input.js
Comment (tomzemp) [reply components/select/src/single-select/input.js:27]: @Mohammer5: won't this be okay though in the sense that now @cooper-joe is using the native <button> element rather than our <Button> component in <InputClearButton> ? We don't seem to allow users to pass their own `onClear` down, so this seems like it would be fine to me?

**Review by tomzemp — APPROVED** [review]

File: components/select/src/single-select/input.js
Comment (Mohammer5) [reply components/select/src/single-select/input.js:27]: I can't follow. What I can see, is:

* The select's input component defines the `onClear` function, which is called when the user clicks the clear button.
* The `onClear` function will call the `onChange` callback provided by the user
* With the currently proposed implementation, the second parameter wouldn't be an event

Comment (cooper-joe) [reply components/select/src/single-select/input.js:27]: I was under the impression that this was OK, as @tomzemp says the button is now a `<buttton>` rather than a `<Button>`.

Comment (Mohammer5) [reply components/select/src/single-select/input.js:27]: Ah yes, my bad! When I used GH to go to the definition, it sent me to the master branch... Sorry!

**Pull request merged**
